Exercise_nonsequential/main.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# Bash script to list the number of sub-folders starting with the
+# keyword 'sub' located inside a working directory. It then submits a
+# Slurm job as an array job of array size equal to the number of 'sub'
+# sub-folders along with a batch script to process those folders.
+
+# Usage:
+# ./main.sh
+
+# Change 'WORKDIR' based on desired location
+WORKDIR=$PWD/masks
+
+ENTRIES=$(ls $WORKDIR | grep sub | wc -l)
```

Comment:
If you are looking for directories starting with "sub", we can also do
... grep ^sub ...

Reply:
@ShobanaSekar , yes, you are right one can certainly implement ^sub to ensure that only those entries get selected that have the name starting with "sub". For the directory structure I was working with 'grep sub' worked equally well. But I can modify the script based on this.